src/virtualship/instruments/adcp.py

```diff
@@ -60,19 +64,33 @@ def simulate_adcp(
     # outputdt set to infinite as we just want to write at the end of every call to 'execute'
     out_file = particleset.ParticleFile(name=out_path, outputdt=np.inf)
 
-    for point in sample_points:
-        particleset.lon_nextloop[:] = point.location.lon
-        particleset.lat_nextloop[:] = point.location.lat
-        particleset.time_nextloop[:] = fieldset.time_origin.reltime(
-            np.datetime64(point.time)
-        )
-
-        # perform one step using the particleset
-        # dt and runtime are set so exactly one step is made.
-        particleset.execute(
-            [_sample_velocity],
-            dt=1,
-            runtime=1,
-            verbose_progress=False,
-            output_file=out_file,
-        )
+    # filter out Parcels logging messages
+    for handler in external_logger.handlers:
+        handler.addFilter(Filter())
+
+    # try/finally to ensure filter is always removed even if .execute fails (to avoid filter being appled universally)
+    # also suits starting and ending the rotator for custom log message
+    try:
+        rotator.start()
+
+        for point in sample_points:
+            particleset.lon_nextloop[:] = point.location.lon
+            particleset.lat_nextloop[:] = point.location.lat
+            particleset.time_nextloop[:] = fieldset.time_origin.reltime(
+                np.datetime64(point.time)
+            )
+
+            # perform one step using the particleset
+            # dt and runtime are set so exactly one step is made.
+            particleset.execute(
+                [_sample_velocity],
+                dt=1,
+                runtime=1,
+                verbose_progress=False,
+                output_file=out_file,
+            )
+
+    finally:
+        rotator.stop()
+        for handler in external_logger.handlers:
+            handler.removeFilter(handler.filters[0])
```

Comment:
In Python, this `try` `finally` pattern is normally done via context managers. This means that if something fails in the block of code, the code is torn down after.

This will simplify both the code for the spinner, as well as the code for the filtering, boiling down to something like

```py
with spinner("Deploying drifters..."):
    with discard_parcels_warnings():
        # code to process measurement
```